src/index.ts

```diff
@@ -1,51 +1,68 @@
 import { compactDecrypt, importJWK, importPKCS8 } from 'jose'
-import {
-  Client,
-  ClientAuthMethod,
-  generators,
-  Issuer,
-  ResponseType,
-} from 'openid-client'
+import { Client, generators, Issuer } from 'openid-client'
 
+import {
+  API_VERSION,
+  DEFAULT_SCOPE,
+  DEFAULT_SGID_CODE_CHALLENGE_METHOD,
+  SGID_AUTH_METHOD,
+  SGID_SIGNING_ALG,
+  SGID_SUPPORTED_FLOWS,
+} from './constants'
 import { convertPkcs1ToPkcs8 } from './util'
 
-const SGID_SIGNING_ALG = 'RS256'
-const SGID_SUPPORTED_FLOWS: ResponseType[] = ['code']
-const SGID_AUTH_METHOD: ClientAuthMethod = 'client_secret_post'
+type ConstructorParams = {
+  clientId: string
+  clientSecret: string
+  privateKey: string
+  redirectUris?: string[]
+  hostname?: string
+}
+
+type AuthorizationUrlParams = {
+  state: string
+  scope?: string | string[]
+  nonce?: string | null
+  redirectUri?: string
+  codeChallenge: string
+}
+
+type AuthorizationUrlReturn = { url: string; nonce?: string }
+
+type CallbackParams = {
+  code: string
+  nonce?: string | null
+  redirectUri?: string
+  codeVerifier: string
+}
 
 export class SgidClient {
+  private apiVersion: number
```

Comment:
in the previous implementation (where access to v1 and v2 coexists in the SDK), a check was being done to ensure that the `apiVersion` must be `2` when constructing an authorization url with the `codeChallenge`. 

that has since been removed and thus `apiVersion` is now only being used in the constructor. i'll remove this attribute now as part of the cleanup - thanks!